### PR TITLE
Autofill password manager integration

### PR DESCRIPTION
**Pull request by darkdh**

### CODE REVIEW

**Comment by darkdh** [comment]
working on unresolved symbols on windows

**Comment by darkdh** [comment]
working on password store on linux

**Comment by darkdh** [comment]
All platforms works now, please review thx.

File: brave/browser/BUILD.gn [hunk 1]
```diff
@@ -25,6 +45,26 @@ source_set("browser") {
     "guest_view/brave_guest_view_manager_delegate.cc",
     "notifications/platform_notification_service_impl.h",
     "notifications/platform_notification_service_impl.cc",
+    "password_manager/brave_password_manager_client.h",
```
Comment (bridiver) [inline brave/browser/BUILD.gn]: all //chrome sources should go in chromium_src build files. Please put these in a password manager source set similar to the others in the 2.58.9 branch

File: brave/browser/BUILD.gn [hunk 2]
```diff
@@ -106,6 +152,22 @@ source_set("browser") {
       "//components/version_info:generate_version_info",
     ]
   }
+
+  if (is_linux) {
+    if (use_x11) {
+      sources += [
+        "//chrome/browser/password_manager/password_store_x.h",
```
Comment (bridiver) [inline brave/browser/BUILD.gn]: see above

File: patches/master_patch.patch [hunk 3]
```diff
@@ -231,15 +231,101 @@ index c384b7e054b07fa3eb20ad6b794bd748fcaf5665..22ca673c3e9a82d3dfa4a6e95a9cf744
...
+-    Profile* profile) {
+-  scoped_refptr<PasswordStore> password_store =
+-      GetForProfile(profile, ServiceAccessType::EXPLICIT_ACCESS);
+-  if (!password_store)
+-    return;
+-  syncer::SyncService* sync_service =
+-      ProfileSyncServiceFactory::GetInstance()->GetForProfile(profile);
+-  net::URLRequestContextGetter* request_context_getter =
+-      profile->GetRequestContext();
++// void PasswordStoreFactory::OnPasswordsSyncedStatePotentiallyChanged(
```
Comment (bridiver) [inline patches/master_patch.patch]: can we stub the deps instead of patching the src?

File: patches/master_patch.patch [hunk 4]
```diff
@@ -231,15 +231,101 @@ index c384b7e054b07fa3eb20ad6b794bd748fcaf5665..22ca673c3e9a82d3dfa4a6e95a9cf744
...
++//                               content::BrowserThread::DB));
++// }
+ 
+ // static
+ void PasswordStoreFactory::TrimOrDeleteAffiliationCache(Profile* profile) {
+@@ -115,7 +115,7 @@ PasswordStoreFactory::PasswordStoreFactory()
+     : RefcountedBrowserContextKeyedServiceFactory(
+           "PasswordStore",
+           BrowserContextDependencyManager::GetInstance()) {
+-  DependsOn(WebDataServiceFactory::GetInstance());
```
Comment (bridiver) [inline patches/master_patch.patch]: can we also stub here?

**Comment by darkdh** [comment]
rebased to 2.58.9 in 0e79af6e8f409a550506e04652be9b5f1fa8d5e7

**Comment by darkdh** [comment]
fixed and verified it works properly on all platforms

File: patches/master_patch.patch [hunk 5]
```diff
@@ -216,21 +216,21 @@ index c384b7e054b07fa3eb20ad6b794bd748fcaf5665..22ca673c3e9a82d3dfa4a6e95a9cf744
 +
```
Comment (bridiver) [inline patches/master_patch.patch]: are there actually any changes here? It doesn't look like it

Comment (darkdh) [inline patches/master_patch.patch]: removed in 95d0f779653aaa85c8ac18aa192666612a21d942

File: chromium_src/BUILD.gn [hunk 6]
```diff
@@ -45,6 +45,25 @@ config("chromium_src_config") {
   }
 }
 
+if (is_win) {
+  # This is in a separate config so the flags can be applied to dependents.
+  # ldflags in GN aren't automatically inherited.
+  config("browser_win_linker_flags") {
```
Comment (bridiver) [inline chromium_src/BUILD.gn]: this doesn't need to be copied, you can just use `//chrome/browser:browser_win_linker_flags` below

Comment (darkdh) [inline chromium_src/BUILD.gn]: removed

**Pull request merged**